internal/resources/cluster.go

```diff
@@ -322,11 +333,22 @@ func (c *Cluster) Update(ctx context.Context, req resource.UpdateRequest, resp *
 		Description: plan.Description.ValueString(),
 		Name:        plan.Name.ValueString(),
 		Support: clusterapi.Support{
-			Plan:     clusterapi.SupportPlan(plan.Support.Plan.ValueString()),
-			Timezone: clusterapi.SupportTimezone(plan.Support.Timezone.ValueString()),
+			Plan: clusterapi.SupportPlan(plan.Support.Plan.ValueString()),
 		},
 	}
 
+	if !plan.Support.Timezone.IsNull() && !plan.Support.Timezone.IsUnknown() {
+		ClusterRequest.Support.Timezone = clusterapi.SupportTimezone(plan.Support.Timezone.ValueString())
+
+		if ClusterRequest.Support.Plan == clusterapi.SupportPlan("basic") && ClusterRequest.Support.Timezone != clusterapi.SupportTimezone("PT") {
+			resp.Diagnostics.AddError(
```

Comment:
Instead of returning an error, why not just set timezone to PT and return a warning ?

Reply:
Can do. It would make it consistent with management api, but shouldn't we discourage users from having a timezone field set for basic?

Reply:
Yes I agree with discouraging users from setting the timezone, hence the warning.

Today the management API allows creating a cluster with basic and any timezone.  If you're suggesting V4 API will be changed to return an error that is a breaking change, which we should not do.

Reply:
As discussed on slack, this will result in error `Provider produced inconsistent result after apply`.  Apologies, the original approach of returning an error is right.

Reply:
Unfortunately, I don't think we can get away with warning due to inconsistency in what gets returned.
Eg:
```
╷
│ Warning: Warning during cluster creation
│ 
│   with couchbase-capella_cluster.new_cluster,
│   on create_cluster.tf line 9, in resource "couchbase-capella_cluster" "new_cluster":
│    9: resource "couchbase-capella_cluster" "new_cluster" {
│ 
│ Timezone field is ignored for basic plan
╵
╷
│ Error: Provider produced inconsistent result after apply
│ 
│ When applying changes to couchbase-capella_cluster.new_cluster, provider "provider[\"registry.terraform.io/couchbasecloud/couchbase-capella\"]" produced an unexpected new value: .support.timezone: was cty.StringVal("GMT"), but now cty.StringVal("PT").
│ 
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
```